pkg/didcomm/protocol/introduce/states.go

```diff
@@ -138,26 +139,32 @@ func (s *arranging) CanTransitionTo(next state) bool {
 }
 
 func (s *arranging) ExecuteInbound(ctx internalContext, m *metaData) (state, error) {
+	// after receiving a response we need to determine whether it is skip proposal or no
+	// if this is skip proposal we do not need to send a proposal to another introducee
+	// we just simply go to Delivering state
+	if m.Msg.Header.Type == ResponseMsgType && isSkipProposal(m) {
+		return &delivering{}, nil
+	}
+
 	if approve, ok := getApproveFromMsg(m.Msg); ok && !approve {
 		return &abandoning{}, nil
 	}
 
-	destinations := m.dependency.Destinations()
+	recipients := fillMissingDestination(m.dependency.Recipients())
 
-	var destination *service.Destination
+	recipient := recipients[len(recipients)-1]
```

Comment:
this could use a comment.

Reply:
along with L158

Reply:
Refactored!
https://github.com/hyperledger/aries-framework-go/pull/946/files#diff-6c72066e0cf828585a690ba919b45ad7R157-R161
https://github.com/hyperledger/aries-framework-go/pull/946/files#diff-6c72066e0cf828585a690ba919b45ad7R234-R235